sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusModelFactory.cs

```diff
@@ -17,6 +17,36 @@ namespace Azure.Messaging.ServiceBus
     /// </summary>
     public static class ServiceBusModelFactory
     {
+        /// <summary>
+        /// Creates a new ServiceBusReceivedMessage instance for mocking.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static ServiceBusReceivedMessage ServiceBusReceivedMessage(
+            BinaryData body,
+            string messageId,
+            string partitionKey,
+            string viaPartitionKey,
+            string sessionId,
+            string replyToSessionId,
+            TimeSpan timeToLive,
+            string correlationId,
+            string subject,
+            string to,
+            string contentType,
+            string replyTo,
+            DateTimeOffset scheduledEnqueueTime,
+            IDictionary<string, object> properties,
+            Guid lockTokenGuid,
+            int deliveryCount,
+            DateTimeOffset lockedUntil,
+            long sequenceNumber,
+            string deadLetterSource,
+            long enqueuedSequenceNumber,
+            DateTimeOffset enqueuedTime) =>
+            ServiceBusReceivedMessage(body, messageId, partitionKey, viaPartitionKey, sessionId, replyToSessionId,
+            timeToLive, correlationId, subject, to, contentType, replyTo, scheduledEnqueueTime, properties,
+            lockTokenGuid, deliveryCount, lockedUntil, sequenceNumber, deadLetterSource, enqueuedSequenceNumber, enqueuedTime);
```

Comment:
Will need to pass in the a default state here otherwise it will keep calling this same method.

Reply:
_sigh_ I should have caught that. That's what happens when you code during short breaks while waiting for the kid...

Reply:
No worries - I mostly caught it due to the CI throwing a stack overflow 😆